shared/chat/hud/mention-hud-container.js

```diff
@@ -14,11 +15,23 @@ type ConnectedMentionHudProps = {
   style?: Object,
 }
 
+const fullNameSelector = inbox => (inbox ? inbox.get('fullNames') : null)
+const participantsSelector = inbox => (inbox ? inbox.get('participants') : null)
+
+const userSelector = createSelector(fullNameSelector, participantsSelector, (fullNames, participants) => {
+  return participants
+    ? participants.reduce((res, username) => {
+        const fullName = fullNames ? fullNames.get(username) : ''
+        res.push({fullName: fullName || '', username})
+        return res
+      }, [])
+    : []
+})
+
 const mapStateToProps: MapStateToProps<*, *, *> = state => {
   const inbox = getSelectedInbox(state)
-  const participants = inbox ? inbox.get('participants').toArray() : []
   return {
-    userIds: participants,
+    users: userSelector(inbox),
```

Comment:
this is fine , you can also just chain the reselect so you pass it state and that calls another selector which pulls out inbox (more typical)